editoast/src/views/work_schedules.rs

```diff
@@ -175,15 +184,107 @@ async fn create(
     }))
 }
 
+#[derive(Serialize, Deserialize, ToSchema)]
+struct WorkScheduleProjectForm {
+    work_schedule_group_id: i64,
+    #[schema(value_type = Vec<TrackRange>)]
+    path_track_ranges: Vec<CoreTrackRange>,
+}
+
+#[derive(Serialize, Deserialize, ToSchema, PartialEq, Debug)]
+struct WorkScheduleProjectResponse {
+    projections: Vec<WorkScheduleProjection>,
+}
+
+#[derive(Serialize, Deserialize, ToSchema, PartialEq, Debug)]
+struct WorkScheduleProjection {
+    #[serde(rename = "type")]
+    pub work_schedule_type: WorkScheduleType,
+    pub start_date_time: NaiveDateTime,
+    pub end_date_time: NaiveDateTime,
+    pub path_position_ranges: Vec<(u64, u64)>,
+}
+
+#[utoipa::path(
+    post, path = "",
+    tag = "work_schedules",
+    request_body = WorkScheduleProjectForm,
+    responses(
+        (
+            status = 201, 
+            body = WorkScheduleProjectResponse, 
+            description = "Returns a list of work schedule whose track ranges intersect the given path"
+        ),
+    )
+)]
+async fn project(
+    State(app_state): State<AppState>,
+    Extension(authorizer): AuthorizerExt,
+    Json(WorkScheduleProjectForm {
+        work_schedule_group_id,
+        path_track_ranges,
+    }): Json<WorkScheduleProjectForm>,
+) -> Result<Json<WorkScheduleProjectResponse>> {
+    let authorized = authorizer
+        .check_roles([BuiltinRole::WorkScheduleRead].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Unauthorized.into());
+    }
+
+    // get all work_schedule of the group
+    let db_pool = app_state.db_pool_v2.clone();
+    let conn = &mut db_pool.get().await?;
+    let settings: SelectionSettings<WorkSchedule> = SelectionSettings::new()
+        .filter(move || WorkSchedule::WORK_SCHEDULE_GROUP_ID.eq(work_schedule_group_id));
+    let work_schedules = WorkSchedule::list(conn, settings).await?;
+
+    let projections = work_schedules
+        .into_iter()
+        .filter_map(|ws| {
+            let ws_track_ranges: Vec<CoreTrackRange> = ws
+                .track_ranges
+                .into_iter()
+                .map(|tr| CoreTrackRange {
+                    track_section: tr.track,
+                    begin: (tr.begin * 1000.0) as u64,
+                    end: (tr.end * 1000.0) as u64,
+                    direction: Direction::StartToStop,
+                })
+                .collect();
+
+            let path_projection = PathProjection::new(&ws_track_ranges);
+            // project this work_schedule on the path
+            let path_position_ranges = path_projection.get_intersections(&path_track_ranges);
```

Comment:
Actually, I have some doubts as to whether `get_intersections` is the function we need.


There's a slight difference between the train schedule projection and the work schedule projection:
- A train schedule path is linear, a work schedule track ranges are not
- A PathProjection required unique tranck sections, which might not be the case for work schedule track ranges

------------------

I think we could adapt the get_intersections function to make it work in the general case (work schedule). This could be done in a later PR.

Reply:
alright, I made a ticket https://github.com/OpenRailAssociation/osrd/issues/8864